tests/msc3391_test.go

```diff
@@ -0,0 +1,274 @@
+//go:build msc3391
+// +build msc3391
+
+// This file contains tests for deleting account data as
+// defined by MSC3391, which you can read here:
+// https://github.com/matrix-org/matrix-doc/pull/3391
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+
+	"github.com/tidwall/gjson"
+)
+
+const testAccountDataType = "org.example.test"
+
+var testAccountDataContent = map[string]interface{}{"test_data": 1}
+
+func TestRemovingAccountData(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	// Create a user to manipulate the account data of
+	aliceUserID := "@alice:hs1"
+	alice := deployment.Client(t, "hs1", aliceUserID)
+
+	// And create a room with that user where we can store some room account data
+	roomID := alice.CreateRoom(t, map[string]interface{}{})
+
+	for _, httpMethod := range []string{"PUT", "DELETE"} {
+		t.Run(fmt.Sprintf("Deleting a user's account data via %s works", httpMethod), func(t *testing.T) {
+			createAndDeleteAccountData(t, alice, httpMethod == "DELETE", nil)
+		})
+		t.Run("Deleting a user's room account data via ", func(t *testing.T) {
+			createAndDeleteAccountData(t, alice, httpMethod == "DELETE", &roomID)
+		})
+	}
+}
+
+func createAndDeleteAccountData(t *testing.T, c *client.CSAPI, viaDelete bool, roomID *string) {
+	// Create the account data and check that it has been created successfully
+	createAccountData(t, c, roomID)
+
+	// Delete the account data and check that it was deleted successfully
+	deleteAccountData(t, c, viaDelete, roomID)
+}
+
+// createAccountData creates some account data for a user or a room, and checks that it was
+// created successfully by both querying the data afterwards, and ensuring it appears down /sync.
+func createAccountData(t *testing.T, c *client.CSAPI, roomID *string) {
```

Comment:
Could this be separate into two functions? It looks like the only duplicated code would be the initial sync?

If you really want to keep a parameter to `createAndDeleteAccountData` for the room ID, I think the if statement could go in there for clarity.